### PR TITLE
Prevent cancelling of in-progress toolshed uploads

### PR DESCRIPTION
**Pull request by wm75**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [x] - This PR does something else (explain below)

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -5,7 +5,9 @@ env:
   GALAXY_BRANCH: release_21.05
   MAX_CHUNKS: 4
 concurrency:
-  group: pr-${{ github.ref }}
+  # group runs by PR, but keep runs on master separate
+  # because we do not want to cancel toolshed uploads
+  group: pr-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:10]:
```suggestion
  group: pr-${{ (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/main') && github.run_number || github.ref }}
```

**Comment by bernt-matthias** [comment]
Wondering how such a case might happen? Is it commits to merged PRs?

**Comment by wm75** [comment]
@bernt-matthias this is about merge commits.
In particular, the case when a merge commit triggered a toolshed upload with preceding tests, but then another merge commit is made.

**Comment by nsoranzo** [comment]
> Wondering how such a case might happen? Is it commits to merged PRs?

I suppose with 2 PRs merged in quick succession, the TS upload of the first one may be cancelled.

**Review by nsoranzo — APPROVED** [review]

**Review by bernt-matthias — APPROVED** [review]

**Comment by wm75** [comment]
Thanks @nsoranzo and @bernt-matthias! So at least the within PR cancelling seems to work still (it just happened over in #3955).

**Pull request merged**